cub/cub/util_device.cuh

```diff
@@ -633,14 +633,65 @@ struct ChainedPolicy
 
   /// Specializes and dispatches op in accordance to the first policy in the chain of adequate PTX version
   template <typename FunctorT>
-  CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE static cudaError_t Invoke(int device_ptx_version, FunctorT& op)
+  CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE static cudaError_t
+  Invoke(int device_ptx_version, FunctorT& op, ::cuda::std::true_type = {})
   {
     if (device_ptx_version < PolicyPtxVersion)
     {
-      return PrevPolicyT::Invoke(device_ptx_version, op);
+      // only continue traversing if the lowest architecture we are compiling for is smaller than the current policy
+      // TODO(bgruber): replace dispatching by if constexpr in C++17
+      constexpr bool cond =
+#ifdef __CUDA_ARCH_LIST__
+        get_first({__CUDA_ARCH_LIST__}) < PolicyPtxVersion;
+#else
+        true;
```

Comment:
Should this be `false`?

Reply:
`true` retains the old behavior (of calling the next policy), which is what we should do when `__CUDA_ARCH_LIST__` is not defined. But given the confusion, I proposed better names for the condition.